internal/cmd/egctl/status.go

```diff
@@ -22,13 +22,28 @@ import (
 	gwv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 
 	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
 )
 
-// supportedTypes list all the resource types that status command supports.
-var supportedTypes = []string{
-	"GatewayClass", "Gateway", "HTTPRoute", "GRPCRoute",
-	"TLSRoute", "TCPRoute", "UDPRoute", "BackendTLSPolicy",
-	"BackendTrafficPolicy", "ClientTrafficPolicy", "EnvoyPatchPolicy", "SecurityPolicy",
+var (
+	supportedXRouteTypes = []string{
```

Comment:
lets bring this up in the community meeting to get consensus on whether to introduce `xRoute` and `xPolicy` here or not

Reply:
The idea of implementing `xRoute` and `xPolicy` for `egctl x status` is actually from #2028 (the `Managed Resources Status` section).

Reply:
resolving this one, should be fine to add this in